components/angular-tomitribe-fab/src/tomitribe-fab.ts

```diff
@@ -96,17 +103,30 @@ module tomitribe_fab {
                         //Arrow down || Arrow right || Enter -> Open
                         setMenuStatus(true, $event);
 
-                    } else if (isOpen() && (($event.keyCode === 40) || ($event.keyCode === 9 && !$event.shiftKey && (isFocusOnTrigger() || hasNextFocusableElement())))) {
+                    } else if (isOpen() && (($event.keyCode === 40) || ($event.keyCode === 9 && !$event.shiftKey))) {
                         //Arrow down || TAB -> Navigate down through menu items
-                        if (isFocusOnTrigger()) {
-                            //reset index
-                            scope.selectedIndex = 0;
+                        if(isFocusOnTrigger() || hasNextFocusableElement()) {
+                            if (isFocusOnTrigger()) {
+                                //reset index
+                                scope.selectedIndex = 0;
+                            }
+                            moveDown($event);
+                        } else {
+                            //Going to exit
+                            if(scope.openOnFocus) {
+                                setMenuStatus(false, $event, true);
```

Comment:
I believe going to exit is not needed here as we are waiting for user to click esc or enter...
WDYT?

Reply:
I tested this case, and we need to close the menu.

Reply:
Ok, was mistaken about one thing.